perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow_template.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing class for GCP's Dataflow service.
+Use this module for running Dataflow jobs from pre-built Dataflow templates
+such as https://cloud.google.com/dataflow/docs/guides/templates/provided-templates
+
+No Clusters can be created or destroyed, since it is a managed solution
+See details at: https://cloud.google.com/dataflow/
+"""
+
+import os
+import re
+import time
+import json
+import logging
+import datetime
+
+from absl import flags
+from perfkitbenchmarker import beam_benchmark_helper
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataflow
+from perfkitbenchmarker.providers.gcp import util
+
+# Refer to flags with prefix 'dpb_df_template' defined in gcp provider flags.py
+FLAGS = flags.FLAGS
+
+class GcpDpbDataflowTemplate(gcp_dpb_dataflow.GcpDpbDataflow):
```

Comment:
Rename to GcpDpbDataflowStreamingTemplate, we might add in the future support for batch templates.

Reply:
Same dataflow template provider can be used to run [batch or streaming template](https://cloud.google.com/dataflow/docs/guides/templates/provided-templates), which is why I named it as such without specificity. However, I can see us extend GetCompletedJob to add support for different input types (PubSub, BQ, GCS)...I'll add a TODO there. SG?

Reply:
My point for making this class explicitly for streaming would be that you have to add logic to check for the stream's end. Whereas if you run a batch, it just ends. I'd rather have all the complexity of dealing with the "Stream end" on its own separate class.